openedx_events/enterprise/signals.py

```diff
@@ -0,0 +1,23 @@
+"""
+Custom signals definitions that representing the Open edX platform events.
+
+All signals defined in this module must follow the name and versioning
+conventions specified in docs/decisions/0002-events-naming-and-versioning.rst
+
+They also must comply with the payload definition specified in
+docs/decisions/0003-events-payload.rst
+"""
+
+from openedx_events.enterprise.data import LicenseLifecycle
+from openedx_events.tooling import OpenEdxPublicSignal
+
+# .. event_type: org.openedx.enterprise.subscription.license.modified.v1
+# .. event_name: LICENSE_CREATED
+# .. event_description: emitted with Subscription.License is created.
+# .. event_data: LicenseLifecycle
+LICENSE_CREATED = OpenEdxPublicSignal(
+    event_type="org.openedx.enterprise.subscription.license.created.v1",
```

Comment:
Given the "warning" I asked to have added in a separate comment, I don't think this comment is a blocker, but I'd like to start a discussion here.
1. The "lifecycle" part of the data was moved to the event_type and name. That is why I am proposing it shouldn't really be in the Data object name. At least, according to patterns established by things like course enrollment events: https://github.com/eduNEXT/openedx-events/blob/main/openedx_events/learning/signals.py#L38-L59
2. We need to research a bit more about how to handle the translation of this to both topics and consumer. The book we were reading was recommending one event type per topic, and one topic per consumer. During book club discussion, for lifecycle events, some people were assuming that we might follow that pattern, but that "created" vs "revoked" etc. would be part of the data, which would make it simpler to ensure proper ordering, rather than working across topics. Also, handling multiple lifecycle topics across multiple consumers would add even more complexity. This may not be clear, other than to say we need to spend more time on the best pattern for this.
3. Lastly, is it possible we may want to have a different pattern for these event bus events than for signals? For example with enrollments? If so, what are the implications for what we are designing?

As stated above, I don't want this comment to be a blocker, but I think it will need more discovery/conversations. It's great that this PR can kickstart this work.

Reply:
@jinder1s: I'm going to move 2 and 3 into a discovery ticket.